server/src/main/java/org/elasticsearch/search/dfs/DfsPhase.java

```diff
@@ -215,6 +215,13 @@ static DfsKnnResults singleKnnSearch(Query knnQuery, int k, Profilers profilers,
                 CollectorResult.REASON_SEARCH_TOP_HITS
             );
             topDocs = searcher.search(knnQuery, ipcm);
+
+            if (knnQuery instanceof ProfilingKnnFloatVectorQuery) {
```

Comment:
You could consider a minor optimization here, to create a `ProfilingKnnVectorQuery` that defines `getVectorOpsCount()` that the specific byte and float classes could inherit from. This could also potentially reduce duplication in `mergeLeafResults()`

Reply:
Indeed, I had considered this approach. However, it entails modifying the [lucene](https://github.com/apache/lucene) codebase since the relevant classes (org.apache.lucene.search.KnnByteVectorQuery and org.apache.lucene.search.KnnFloatVectorQuery) are housed within lucene.